### ISSUE TITLE
Enhance REST API by a technical response with information about the ets 

### ISSUE THREAD

**Issue by lgoltz** [opengeospatial/teamengine#397]
To enable external or other ets the integration of a test suite we should enhance the REST API by a more technical  view. 

## Overview of all available test suites
 * http://cite.opengeospatial.org/te2/rest/suites/
 * with Content-Type `application/xml`
 * should return a well-formed and valid (schema must be created) xml document listing the available test suites 

## Overview of all available test suites
 * http://cite.opengeospatial.org/te2/rest/suites/wfs20
 * with Content-Type `application/xml`
 * should return a well-formed and valid (schema must be created) xml document listing the parameters of the wfs20 REST API

**Comment by dstenger** [comment]
Idea: Introduce Swagger.

**Comment by ghobona** [comment]
Ideally, this should return JSON with Content Type `application/json`.

**Comment by ghobona** [comment]
Agreed with @dstenger on 2020-06-23 that content negotiation should be through the HTTP Accepted header. 

Format options:

* JSON
* HTML
* XML

**Comment by keshav-nangare** [comment]
Please find the below responses for the `Available Test Suites` REST API with content negotiation and the client can get a response based on the request header.
  1. **Accept : application/xhtml+xml**
Response:
```
<?xml version="1.0" encoding="UTF-8"?>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
        <title>Available test suites</title>
    </head>
    <body>
        <h1>Available test suites</h1>
        <ul>
            <li>
                <a href="gml32/" id="gml32" type="text/html">GML (ISO 19136:2007) Conformance Test Suite, Version 3.2.1</a>
            </li>
            <li>
                <a href="wms13/" id="wms13" type="text/html">Conformance Test Suite - OGC Web Map Service 1.3.0</a>
            </li>
        </ul>
    </body>
</html>
```
2. **Accept : application/xml**
Response:
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
    <testsuite>
        <title>GML (ISO 19136:2007) Conformance Test Suite, Version 3.2.1</title>
        <resturi>http://localhost:8080/teamengine/rest/gml32/</resturi>
    </testsuite>
    <testsuite>
        <title>Conformance Test Suite - OGC Web Map Service 1.3.0</title>
        <resturi>http://localhost:8080/teamengine/rest/wms13/</resturi>
    </testsuite>
</testsuites>
```
3. **Accept : application/json**
Response:
```
{
    "testsuites": [
        {
            "title": "Conformance Test Suite - OGC Web Map Service 1.3.0",
            "resturi": "http://localhost:8080/teamengine/rest/wms13/"
        },
        {
            "title": "GML (ISO 19136:2007) Conformance Test Suite, Version 3.2.1",
            "resturi": "http://localhost:8080/teamengine/rest/gml32/"
        }
    ]
}
```

@dstenger @ghobona Please let me know your thoughts on this format of responses.

**Comment by ghobona** [comment]
@keshav-nangare 

I suggest the following:

```json
{
    "testsuites": [
        {
            "title": "Conformance Test Suite - OGC Web Map Service 1.3.0",
            "testSuiteVersions": "1.25",
            "status": "final",
            "resturi": "http://localhost:8080/teamengine/rest/wms13/"
        },
        {
            "title": "GML (ISO 19136:2007) Conformance Test Suite, Version 3.2.1",
            "testSuiteVersions": "1.27",
            "status": "final",
            "resturi": "http://localhost:8080/teamengine/rest/gml32/"
        }
    ]
}

```

The values would be consistent with those presented on the TEAM Engine homepage.

![Screenshot 2020-06-29 at 15 55 32](https://user-images.githubusercontent.com/18121038/86021629-31c67c00-ba21-11ea-808e-a2cba01a201f.png)

**Comment by keshav-nangare** [comment]
@ghobona 

Thanks for inputs!

This table is prepared by using of config.xml file which contains all information. While invoking the REST API we are not using the config.xml file so we don't have the scope of config.xml file. 
But we can get this information in a different way. I have investigated the REST API execution so we have the info test suite `title`, `Revision` but don't have the `status` field. If the status field is mandatory then will need to do a different workaround.

Can please confirm do we require status field in response?

**Comment by ghobona** [comment]
@keshav-nangare Ok, you can leave out `status`. Thanks.

**Comment by dstenger** [comment]
**Decision during meeting on July 1st 2020**

**Provided information:**

`status` is removed and `etsCode` is added.

Thus, following elements will be present in XML and JSON encoding:
* title
* version
* endpoint
* etsCode

**Information about single test suite:**

XML file is created which is used to generate all three encodings of the information site about a single test suite.
The old mechanism (HTML template is used) shall still work.

**Comment by ghobona** [comment]
@keshav-nangare showed @ghobona the demo on 2020-07-08

**Comment by keshav-nangare** [comment]
Fixed with #434 PR.
